CMakeLists.txt

```diff
@@ -746,10 +784,15 @@ if (BUILD_TESTING)
   target_include_directories(testmamico PRIVATE ${CPPUNIT_INCLUDE_DIRS})
   target_link_directories(testmamico PRIVATE ${CPPUNIT_LIBRARY})
   target_link_libraries(testmamico PRIVATE cppunit)
-  foreach(TEST ${TEST_SOURCES})
-    get_filename_component(TEST_NAME ${TEST} NAME_WE)
+
+  execute_process(COMMAND cmake -DGET_TEST_NAMES=ON -DBUILD_TESTING=ON -DBUILD_WITH_MPI=${BUILD_WITH_MPI} .)
+  execute_process(COMMAND make getTestNames)
+  execute_process(COMMAND ./getTestNames OUTPUT_VARIABLE TEST_NAMES)
+  separate_arguments(TEST_NAMES UNIX_COMMAND ${TEST_NAMES})
+
+  foreach(TEST_NAME ${TEST_NAMES})
     if (BUILD_WITH_MPI)
-      add_test(NAME ${TEST_NAME} COMMAND mpirun -np 4 ./testmamico ${TEST_NAME})
+      add_test(NAME ${TEST_NAME} COMMAND mpirun --oversubscribe -np 4 ./testmamico ${TEST_NAME})
```

Comment:
`--oversubscribe` is a OpenMPI-specific option. Leave a comment here about that so that if you ever change your docker image to something else (e.g. mpich) you at least know what to change

Reply:
Done d09c4d5c68de4e2cc1347bbcbe14daddacb25672